samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt

```diff
@@ -130,29 +130,31 @@ foreach(GROUP BASE NOTIFICATIONS ALERTS STATELISTENER SPEECHDATA MUSICDATA CUSTO
         get_filename_component(FILE_BASE ${file} NAME_WE)
         list(APPEND PROTOC_INC_DIRS "-I.")
         list(APPEND PROTOC_INC_DIRS "-I${FILE_PATH}")
-        list(APPEND ${GROUP}_PROTO_BYPRODUCTS "${PROTO_COMPILATION_DIR}/${FILE_BASE}.pb.c")
+        list(APPEND ${GROUP}_PROTO_OUTPUTS "${PROTO_COMPILATION_DIR}/${FILE_BASE}.pb.c")
+        list(APPEND ${GROUP}_PROTO_DEPENDENCIES "${file}")
     endforeach()
 
-    # Populate list of compilation commands for protoc compiler.
-    unset(PROTOC_CMD_LIST)
+    # Each .pb.c file to be compiled by protoc gets its own CMake custom command,
+    # with dependencies to all other .proto files in its group and the base files.
     foreach(file ${${GROUP}_PROTO_FILES})
         get_filename_component(FILE_PATH ${file} DIRECTORY)
```

Comment:
Maybe this should have been `FILE_DIR` instead.

Usually `_DIR` is most common in CMake, like `CURRENT_LIST_DIR`, `CURRENT_BINARY_DIR` and so on.

Reply:
Updating name as suggested